### PR TITLE
sogo: Init at 4.0.8

### PR DESCRIPTION
**Pull request by dasJ**
I know this isn't the most beautiful code, but gnustep doesn't really
like the way nix handles the dependencies and would rather write every
library and binary to the same directory.

###### Motivation for this change

I also have a module for this, but that's for a later PR.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/sope/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    ++ optional (postgresql != null) "--enable-postgresql";
+
+  # Yes, this is ugly.
+  preFixup = ''
+    cp -rlPa $out/nix/store/*/* $out
+    rm -rf $out/nix/store
+  '';
+
+  meta = {
+    description = "SOPE is an extensive set of frameworks which form a complete Web application server environment";
```
Comment (mmahut) [inline pkgs/development/libraries/sope/default.nix]: Please do not start with the package name.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -13365,6 +13366,8 @@ in
 
   sonic = callPackage ../development/libraries/sonic { };
 
+  sope = pkgs.callPackage ../development/libraries/sope { };
```
Comment (mmahut) [inline pkgs/top-level/all-packages.nix]: Can you make one commit per new package please?

**Comment by mmahut** [comment]
@GrahamcOfBorg build sogo sope

**Comment by dasJ** [comment]
See #79175

**Pull request closed without merging**